### ISSUE TITLE
highlight point on cumulativeLineChart

### ISSUE THREAD

**Issue by xHeliotrope** [krispo/angular-nvd3#295]
I'm curious if there is a way to add a circle to a specific point on a cumulative chart. in most of the d3 tutorials, i've seen that you can do a d3.select("chart svg") and change the attrs of a specific point, but so far I haven't been able to find the specific node that I would need to do so with this library.

I'm looking for something like they did here http://jsfiddle.net/rgg7t/25/, but with this library.

**Comment by krispo** [comment]
We can use `callback` under initialization and dispatch `renderEnd` event after window resizing for point highlighting.
Check [demo](http://plnkr.co/edit/WfcVzs?p=preview).

**Comment by xHeliotrope** [comment]
thanks! I just tried this, and it does not work for me outside of the plunker. it runs the callback highlightPoint before the renderEnd event, and so the dot appears but does not stay. I think it is because my chart is already inside of a .then() from a promise itself. ...anyways, thank you for showing me how this can be done.

**Comment by xHeliotrope** [comment]
found out this is happening because onReady never gets called.

**Comment by krispo** [comment]
check the version, it's fixed in 1.0.5

**Comment by xHeliotrope** [comment]
yea I just updated to 1.0.5 a few minutes ago.. same behavior. I noticed my angular is 1.4.7, and yours is 1.3.9, might that have something to do with it? I am also getting a 'cannot read property xAxis of undefined..' in console, on the second time it goes to call highlight function.

**Comment by krispo** [comment]
How do you define `onReady` function in js and html? Try to define with `dot` notation, or with `data-on-ready`..

**Comment by xHeliotrope** [comment]
got it. I needed $scope.chart; and var chart = $scope.chart, I'm thinking because callback is internal scope? new to angular & d3 so I don't know.

thank you for your help!

**Comment by xHeliotrope** [comment]
hi again. Sorry to bother you, but [here](http://plnkr.co/edit/DK70oepfTe1mvb5MYbFA?p=preview) is the issue I was having. I believe it is the fact that callback has internal scope, so the chart is gone by the time renderEnd is run, due to my service? Its just frustrating because it the points get highlight, and then disappear.

Thanks for your help already.

**Comment by krispo** [comment]
It's because you try to initialize directive attributes after html page has already rendered. In this case we can either generate new directive template with new internal scope after service succeed, or as usual initialize directive with empty data at the beginning and then populate data after service succeed. Modified example is  [here](http://plnkr.co/edit/47MVej?p=preview) (with the second variant).

**Comment by blindstuff** [comment]
Hi guys,

I know this sounds idiotic, but while the solution posted here works, I can't manage to do it on multiple datasets that are defined dinamically. I realize the issue is in this section of the code:

```
      var points = d3.select('.nv-groups')
        .selectAll("circle.myPoint")
        .data(data[0].values.filter(function (d) {
          return d.x === ($scope.setups[0].sag() | 0);
        }));

      points.enter().append("circle")
        .attr("class", "myPoint")
        .attr("cx", function (d) {
          return chart.xAxis.scale()(d.x);
        })
        .attr("cy", function (d) {
          return chart.yAxis.scale()(d.y);
        })
        .attr("r", 4)
        .style("fill", function (d) {
          return d3.select('.nv-series-' + d.series).style('fill');
        });
```

I've tried building a loop around the whole structure or just the the first section to obtain all the points and then give them the necessary styling, but i've not been succesful at it. This is probably more of a d3 questions, but I posted here since its an expansion on the topic at hand.

In short, I'm trying to filter over all X data collections the same way I'm going over the first one in the included code so I can add a styling to points on each of the graph lines.

Edit: Here's is a QA link to what i'm working on. I wan't to be able to draw the fixed dot as seen on 1 of the series lines on all three.

http://suspension-calc-qa.s3-website-us-east-1.amazonaws.com

Thnx
